### PR TITLE
Bug 1821210 -  Added content description on Back button to say `Navigate up button`

### PR DESCRIPTION
**Pull request by gitstart**
### What
Back button when clicked doesn't say "Navigate up button"

### Issue video

https://user-images.githubusercontent.com/39338964/224269698-de088e2b-5f18-4127-b77f-c51fbb5dcbfc.mp4

### Demo video after fix

https://user-images.githubusercontent.com/39338964/224269904-b809f89a-5215-46cc-ba68-8d195917b766.mp4

### Pull Request checklist
 - [x] Quality: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
 - [x] Tests: This PR includes thorough tests or an explanation of why it does not
 - [ ] Changelog: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
 - [ ] Accessibility: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features
### After merge
- Milestone: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- Breaking Changes: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

---
This code was written and reviewed by GitStart Community. Growing future engineers, one PR at a time.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1821210

### CODE REVIEW

**Comment by t-p-white** [comment]
bors try

File: android-components/components/browser/menu/src/main/java/mozilla/components/browser/menu/item/BackPressMenuItem.kt [hunk 1]
```diff
@@ -38,6 +42,13 @@ class BackPressMenuItem(
             backPressListener.invoke()
             menu.dismiss()
         }
+        view.accessibilityDelegate = object : AccessibilityDelegate() {
+            override fun onInitializeAccessibilityNodeInfo(host: View, info: AccessibilityNodeInfo) {
+                super.onInitializeAccessibilityNodeInfo(host, info)
+                info.className = Button::class.java.name
+            }
+        }
+        view.contentDescription =  contentDescription
```
Comment (t-p-white) [inline android-components/components/browser/menu/src/main/java/mozilla/components/browser/menu/item/BackPressMenuItem.kt:51]:
```suggestion
        view.contentDescription = contentDescription
```

**Review by t-p-white — CHANGES_REQUESTED** [review]
Please see comment 🙂

**Comment by gitstart** [comment]
Thank you @t-p-white for reviewing our PR. 
We pushed the changes requested. ✔

**Comment by t-p-white** [comment]
bors try

**Comment by t-p-white** [comment]
bors try

**Comment by t-p-white** [comment]
bors try

**Comment by t-p-white** [comment]
bors try

**Review by t-p-white — APPROVED** [review]
@gitstart thank you for your contribution, nice work 👍

**Pull request merged**